bosh/plugin.go

```diff
@@ -4,27 +4,64 @@ import (
 	"encoding/json"
 	"log"
 	"net/http"
+
+	"github.com/cloudfoundry/bosh-agent/jobsupervisor/monit"
 )
 
 type plugin struct {
-	report report
+	report          report
+	monitHTTPClient monit.Client
 }
 
-func NewPlugin(hostname, jobSpecPath string) *plugin {
+func NewPlugin(hostname, jobSpecPath, platform string) (*plugin, error) {
 	report := newReport(hostname)
 
 	if jobSpec, err := loadSpec(jobSpecPath); err != nil {
 		log.Printf("error loading job spec from %q: %v\n", jobSpecPath, err)
+		return nil, err
 	} else {
-		report.AddJob(jobSpec)
+		report.SetInstanceSpecData(jobSpec)
+	}
+
+	var monitHTTPClient monit.Client
+	var err error
+
+	if platform == "ubuntu" {
```

Comment:
1. Since the `bosh_agent` distinguishes between `ubuntu` and `centos` (https://github.com/cloudfoundry/bosh-agent/blob/6b3648f130069424a62bce2f354418322aa975d5/platform/provider.go#L191-L196) as possible values for `platform`, it seemed sensible to have a similar interface.  For most thing, in particular how it talks to monit, turns out `ubuntu` and `centos` use the same code paths.  So I could see collapsing the two into just `linux`.  On the other hand, if you wanted to surface richer networking information like https://github.com/cloudfoundry/bosh-agent/blob/6b3648f130069424a62bce2f354418322aa975d5/platform/net/ubuntu_net_manager.go#L69 and https://github.com/cloudfoundry/bosh-agent/blob/6b3648f130069424a62bce2f354418322aa975d5/platform/net/ubuntu_net_manager.go#L191, then you'd need to distinguish between the two.

I'd say for now, we could either collapse it to `linux`, or keep the options explicitly different but add to this line with `if platform == "ubuntu" || platform == "centos"`, and likewise in one or two other places.  What do you think?

2. Interesting.  Depends on what we want to do with regards to linux vs ubuntu+centos.

3. No monit on Windows.